server/models/Emoji.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  InferAttributes,
+  InferCreationAttributes,
+  type SaveOptions,
+} from "sequelize";
+import {
+  BeforeCreate,
+  BelongsTo,
+  Column,
+  DataType,
+  ForeignKey,
+  Table,
+} from "sequelize-typescript";
+import { EmojiValidation } from "@shared/validations";
+import { ValidationError } from "@server/errors";
+import Team from "./Team";
+import User from "./User";
+import IdModel from "./base/IdModel";
+import Fix from "./decorators/Fix";
+import IsUrlOrRelativePath from "./validators/IsUrlOrRelativePath";
+import Length from "./validators/Length";
+
+@Table({ tableName: "emojis", modelName: "emoji" })
+@Fix
+class Emoji extends IdModel<
+  InferAttributes<Emoji>,
+  Partial<InferCreationAttributes<Emoji>>
+> {
+  @Length({
+    max: EmojiValidation.maxNameLength,
+    msg: `emoji name must be  less than ${EmojiValidation.maxNameLength} characters`,
+  })
+  @Column(DataType.STRING)
+  name: string;
```

Comment:
I think we'll need some additional validations to ensure that there are no special characters, something like `[a-z0-9-_]`